tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -6356,7 +6356,7 @@ public function dataLoopVariables(): array
 				"'afterAssign'",
 			],
 			[
-				'LoopVariables\Foo',
+				'LoopVariables\Foo|LoopVariables\Lorem',
```

Comment:
This is a good test case to focus on other issues. In the following example, `$foo` should always be `Foo`.

```php
foreach (items() as $item) {
    $foo = new Foo();

    if (condition()) {
        $foo = new Lorem();
        continue;
    }

    $foo; # => Foo, not Foo|Lorem
}
```

However, in order to support this case, we have to check whether `$foo` is always assigned in the loop. Since the current implementation only has the scope in the loop, we must allow `lookForAssignsInBranches` to receive the outer scope. The following is an example.

```php
	private function lookForAssignsInBranches(
		Scope $initialScope,
		array $statementsLists,
		LookForAssignsSettings $lookForAssignsSettings,
		int $counter = 0,
        ?Scope $outerScope # Add this scope
	): Scope
```

But I'm wondering if this change will be accepted... 🤔

Reply:
Please add the fix as you have it in mind as a new commit so I can evaluate it separately 😊

Reply:
Thanks! I'm going to work on this later 👍